OpenRA.Mods.D2k/Traits/Sandworm.cs

```diff
@@ -82,11 +82,12 @@ void RescanForTargets(Actor self)
 
 			// If close enough, we don't care about other actors.
 			var target = self.World.FindActorsInCircle(self.CenterPosition, WormInfo.IgnoreNoiseAttackRange)
-				.FirstOrDefault(x => attackTrait.HasAnyValidWeapons(Target.FromActor(x)));
-			if (target != null)
+				.Select(t => Target.FromActor(t))
+				.FirstOrDefault(t => attackTrait.HasAnyValidWeapons(t));
```

Comment:
The thing to remember about LINQ is that it defers evaluations as late as possible - if the first `HasAnyValidWeapons` is true for the first actor, then the Select will only generate that one target.